libraries/RC_Channel/RC_Channel.cpp

```diff
@@ -1879,7 +1881,21 @@ bool RC_Channel::do_aux_function(const AuxFuncTrigger &trigger)
     }
 #endif
 
-    // do nothing for these functions
+#if AP_SCRIPTING_ENABLED
+    case AUX_FUNC::KILL_SCRIPTING: {
+        AP_Scripting *scr = AP::scripting();
+        if (scr != nullptr) {
+            if (ch_flag == AuxSwitchPos::HIGH) {
+                scr->stop();
+            } else {
+                scr->restart_all();
+            }
```

Comment:
Please `switch` on `ch_flag` instead and do nothing with the middle position.

Reply:
Fixed